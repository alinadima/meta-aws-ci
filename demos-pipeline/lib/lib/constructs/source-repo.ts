import * as cdk from "aws-cdk-lib";
import { Construct } from "constructs";
import { Code, Repository } from "aws-cdk-lib/aws-codecommit";

import * as path from "path";

export enum RepoKind {
  /** A Poky Based Repository */
  Poky = "poky",
  MetaAwsDemo = "meta-aws-demo",
}

export interface SourceRepoProps extends cdk.StackProps {
  readonly repoName: string;
  readonly kind: RepoKind;
}

/**
 * The repository for the Source Stage of the pipeline.
 */
export class SourceRepo extends Construct {
  /** The CodeCommit Repo itself. */
  readonly repo: Repository;

  constructor(scope: Construct, id: string, props: SourceRepoProps) {
    super(scope, id);

    this.repo = new Repository(this, "SourceRepository", {
      repositoryName: props.repoName,
      code: Code.fromDirectory(
        path.join(__dirname, "..", "..", "source-repo", props.kind),
        "main"
      ),
    });
  }
}
