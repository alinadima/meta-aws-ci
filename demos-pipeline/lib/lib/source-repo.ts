import * as cdk from "aws-cdk-lib";
import { Construct } from "constructs";
import { Code, Repository } from "aws-cdk-lib/aws-codecommit";

import * as path from "path";

export enum RepoKind {
  /** A Poky Based Repository */
  Poky = "poky",
}

export interface SourceRepoProps extends cdk.StackProps {
  readonly repoName: string;
  readonly kind: RepoKind;
}

/**
 * The repository for the Source Stage of the pipeline.
 */
export class SourceRepo extends cdk.Stack {
  /** The CodeCommit Repo itself. */
  readonly repo: Repository;

  constructor(scope: Construct, id: string, props: SourceRepoProps) {
    super(scope, id, props);

    this.repo = new Repository(this, "SourceRepository", {
      repositoryName: props.repoName,
      code: Code.fromDirectory(
        path.join(__dirname, "..", "source-repo", props.kind),
        "main"
      ),
    });
  }
}
