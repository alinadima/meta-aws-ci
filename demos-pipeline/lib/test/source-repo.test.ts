import * as cdk from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { SourceRepo, DistributionKind } from "../lib/constructs/source-repo";

describe("Demo Source Repository", () => {
  const props = {
    env: { account: "12341234", region: "eu-central-1" },
    kind: DistributionKind.Poky,
    repoName: "charlie",
  };

  test("Snapshot", () => {
    const app = new cdk.App();
    const stack = new cdk.Stack(app, "TestStack", props);
    new SourceRepo(stack, "MyTestStack", props);
    const template = Template.fromStack(stack);
    expect(template).toMatchSnapshot();
  });
});
