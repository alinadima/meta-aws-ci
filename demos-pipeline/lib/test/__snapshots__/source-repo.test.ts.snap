// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo Source Repository Snapshot 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "MyTestStackSourceRepository1F6C3AB0": {
      "Properties": {
        "Code": {
          "BranchName": "main",
          "S3": {
            "Bucket": "cdk-hnb659fds-assets-12341234-eu-central-1",
            "Key": "be6ddd2184e42352f78f888697689395d48c33c2e70b76fb33f225cd85cd6224.zip",
          },
        },
        "RepositoryName": "charlie",
      },
      "Type": "AWS::CodeCommit::Repository",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
